docs/network-troubleshooting.adoc

```diff
@@ -0,0 +1,143 @@
+
+```
+failed to get unicast channel with peer [peer_id] because of: [cause]
+```
+
+=== Failed to announce signer presence
+
+Announce protocol is performed before the key generation process. It's purpose
+is to exchange public keys between members. Public keys are then used to
+derive the peers transport identifiers necessary to open unicast channels.
+Announce protocol duration is capped with a 2 minutes timeout. In case the
+timeout passes and the client doesn't receive public keys from all of other
+keep members, the following error will be printed in logs:
+
+```
+failed to announce signer presence: [waiting for announcements timed out after: [2m0s]]"
+```
+
+=== Readiness signaling protocol failed
+
+Readiness protocol is performed before both key generation and signing
+processes. It's purpose is to synchronize keep members and allows each member
+to signal their readiness to perform the TSS protocol. Like the announce
+protocol, a 2 minutes timeout applies here as well. In case the timeout
+is exceeded and the client doesn't receive readiness notifications from
+all of other keep members, the following log error is produced:
+
+```
+readiness signaling protocol failed: [waiting for readiness timed out after: [2m0s]]"
+```
+
+== Troubleshooting checklist
+
+The cause of aforementioned problems may lie in either member of the keep.
+To make sure your client performs well, please pass through the following
+checklist.
+
+=== Check your client health
+Check your client is up and doesn't experience frequent restarts. If your clients
+goes down, it breaks the ongoing protocol execution for other keep members.
+This is especially relevant for cloud workloads where things like VMs preemption
+or Kubernetes pod relocation may take place. It's strongly recommended to enable
+<<./run-keep-ecdsa.adoc#Metrics, metrics>>
+for your client and use a monitoring system which allows catching all issues quickly
+(see for one fairly complete example of such a monitoring system:
+ https://github.com/boar-network/keep-monitoring[Boar Network monitoring]).
+Using monitoring tools specific for your cloud provider may also be a good choice.
+
+=== Check internet connectivity
+Check your client has a working and stable internet connection. Temporary
+outages may make your client invisible for other keep members. Other members
+won't be able to connect your client and perform protocol executions. Metrics
+can help monitor network connectivity as they return the number of connected
+peers. A significant fall of connected peers number may indicate network
+issues.
+
+=== Check Ethereum connectivity
+Check if your node has a working connection with their ETH node. Broken
+ETH connection means that your client becomes blind for all chain events
+taking place for keeps it is a member of. For example, it may not notice a
+starting key generation or signing process. ETH connectivity can be
+watched using metrics and a monitoring system
```

Comment:
I would also mention  there should be no errors in the logs about the state of websocket connection.

Reply:
Done https://github.com/keep-network/keep-ecdsa/pull/838/commits/248102e9d48a0600e1d80d5cf1c769832bfce224